src/promptflow-core/promptflow/_core/tool_meta_generator.py

```diff
@@ -533,9 +533,14 @@ def generate_flow_meta_dict_by_file(data: dict, source: str = None, path: str =
         flow_meta[meta_key] = {}
         for k, v in ports.items():
             # We didn't support specifying multiple types for inputs/outputs/init, so we only take the first one.
-            flow_meta[meta_key][k] = {"type": v.type[0].value}
+            param_type = v.type[0]
```

Comment:
With this change, inputs/outputs could also be connections?

Reply:
Or you can say, previously a runtime system error will be raised here if input/output ports are of type connections here, which is not the responsibility of this function.

For this feature (pf flow save), an extra validation will be done via FlexFlowSchema to check the exact types of inputs & outputs. Given this function won't work for dag flow any more and SDK will always do a `pf.flows._save` before sending flex flow to executor, this validation should be enough.